Add codes.merritt.FeelingFinder

Merrit opened the pull request.

hfiguiere commented:
there is no manifest.

Merrit commented:
Right, added the wrong file. That one's the manifest now, I believe.

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,31 @@
+    sources:
+      - type: file
+        path: FeelingFinder-Linux-Portable.tar.gz
+      - type: file
+        path: packaging/linux/flatpak/build-flatpak.sh
+      - type: file
+        path: packaging/linux/flatpak/codes.merritt.FeelingFinder.metainfo.xml
+      - type: file
+        path: packaging/linux/codes.merritt.FeelingFinder.desktop
+      - type: file
+        path: assets/images/icon.svg
```
none of these files are present.

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,31 @@
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        path: FeelingFinder-Linux-Portable.tar.gz
```
you are supposed to download from upstream.

Merrit replied:
I've obviously somewhat misunderstood how it is supposed to work.

The application is already being built by CI, I was trying not to duplicate that work and just have it packaged into a flatpak that could be loaded to flathub. I was referencing in part the OBS workflow: https://github.com/obsproject/obs-studio/blob/master/.github/workflows/flatpak.yml#L124

Is this feasible? I was able to find precious little about integrating flatpak into CI / CD.

hfiguiere replied:
and how to you think flathub knows where to get it?

barthalion replied:
OBS (also Firefox, few others) are special, but it requires too much handholding currently. It will be simpler if you could upload this tar.gz somewhere (be it S3, or GitHub releases), and reference it in your manifest.

Automating the process required to set up direct publishing is on the roadmap, but it's unlikely to happen within the next 6 months.

Merrit replied:
Wasn't quite sure, though I was hoping something simple like: CI builds app, triggers flathub upload. :woman_shrugging: 

I think I've got it to a working place now though, thank you :+1:

hfiguiere reviewed (changes requested):
the build won't work

hfiguiere commented:
bot, build codes.merritt.FeelingFinder

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,33 @@
+---
+app-id: codes.merritt.FeelingFinder
+runtime: org.gnome.Platform
```
Do you really need the GNOME platform?

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,33 @@
+separate-locales: false
+finish-args:
+  - "--share=ipc"
+  - "--socket=x11"
```
```suggestion
```

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,33 @@
+
+      - type: git
+        url: https://github.com/Merrit/feeling_finder.git
+        branch: main
```
You need to specify a revision and not a branch

Merrit commented:
Hope that looks better, thanks for the tips! :)

hfiguiere commented on codes.merritt.FeelingFinder.yml:
```diff
@@ -0,0 +1,33 @@
+      - "./build-flatpak.sh"
+    sources:
+      - type: file
+        url: https://github.com/Merrit/feeling_finder/releases/latest/download/FeelingFinder-Linux-Portable.tar.gz
```
this is not resolved.

barthalion replied:
see https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

hfiguiere replied:
Also the build fail because "latest" is probably pointing to something that change all the time.

You should point to a fixed revision like https://github.com/Merrit/feeling_finder/releases/download/v0.2.0/FeelingFinder-Linux-Portable.tar.gz

barthalion commented:
bot, build codes.merritt.FeelingFinder

barthalion commented:
```
Failed to download sources: module feeling_finder: Wrong sha256 checksum for FeelingFinder-Linux-Portable.tar.gz, expected "467172df32834ccf51136f1a94b4f2ecb7b9c1cab00c5d93dd931882a2cbe999", was "40901ab4666901132d868b11a65193b3b32fd94c7883d03dacc207954617fb4e"
```

Merrit commented:
Failed because of a new release since the manifest was updated, working now.

hfiguiere commented:
> Failed because of a new release since the manifest was updated, working now.

You should address a specific release as I commented.

Merrit commented:
> You should address a specific release as I commented.

Yes, I see what you mean! Changed to link a specific release, hopefully that is better :blush:

hfiguiere commented:
bot, build codes.merritt.FeelingFinder

Merrit commented:
Was using `appstreamcli validate` to validate, which apparently does *not* do the same checks as the flatpak version, xml is passing validation now.

The automated build showed a failure for aarch64, even though I added `only-arches: ["x86_64"]`.. not seeing obvious docs for this, but saw references to a flathub.json that specifies that archs, so I added that - hope that is the correct way.

hfiguiere commented:
> Was using `appstreamcli validate` to validate, which apparently does _not_ do the same checks as the flatpak version, xml is passing validation now.
> 
> The automated build showed a failure for aarch64, even though I added `only-arches: ["x86_64"]`.. not seeing obvious docs for this, but saw references to a flathub.json that specifies that archs, so I added that - hope that is the correct way.

There were link to the documentation in the checklist you deleted when creating the PR, requesting you acknowledge having read, that was indicating what to do:
https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

hfiguiere commented:
bot, build codes.merritt.FeelingFinder

Merrit commented:
> There were link to the documentation in the checklist you deleted when creating the PR, requesting you acknowledge having read, that was indicating what to do:
> [flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64](https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64)

I didn't delete a checklist, there was nothing when I created it. Perhaps because I didn't use the github website interface to create the PR? That's frustrating :confused: .. thank you for all the help, must have been frustrating thinking I was blatantly ignoring the docs! I assure you I've spent several days reading about flatpak and going over the documentation 😵‍💫

I'll go over this page you linked as well!
